conans/model/build_info.py

```diff
@@ -41,6 +47,66 @@ def __init__(self):
         self.description = None  # Description of the conan package
         # When package is editable, filter_empty=False, so empty dirs are maintained
         self.filter_empty = True
+        self._deps = OrderedDict()
+
+    @property
+    def libs(self):
+        if self._deps:
+            deps = [v for v in self._deps.values()]
+            deps_sorted = sorted(deps, key=lambda component: len(component.deps))
+            self._libs = [dep.lib for dep in deps_sorted if dep.lib is not None]
+        return self._libs
+
+    @libs.setter
+    def libs(self, libs):
+        if self._deps:
+            raise ConanException("Setting first level libs is not supported when Components are "
+                                 "already in use")
+        self._libs = libs
+
+    @property
+    def exes(self):
+        if self._deps:
+            deps = [v for v in self._deps.values()]
+            return [dep.exe for dep in deps if dep.exe]
+        else:
+            return self._exes
+
+    @exes.setter
+    def exes(self, exes):
+        if self._deps:
+            raise ConanException("Setting first level exes is not supported when Components are "
+                                 "already in use")
+        self._exes = exes
+
+    @property
+    def system_deps(self):
+        if self._deps:
+            deps = [v for v in self._deps.values()]
+            deps_sorted = sorted(deps, key=lambda component: len(component.deps))
```

Comment:
`self._deps.values()` is already and iterable, no need to build the list in the previous line